subprojects/base-services/src/main/java/org/gradle/util/GUtil.java

```diff
@@ -367,7 +368,7 @@ public static String toConstant(CharSequence string) {
         if (string == null) {
             return null;
         }
-        return toWords(string, '_').toUpperCase();
+        return TextUtil.toUpperCaseUserLocale(toWords(string, '_'));
```

Comment:
I think it's OK to capture the current behavior here, but I bet these are sometimes doing the wrong thing.